src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberFieldSymbol.cs

```diff
@@ -481,7 +481,8 @@ internal sealed override TypeWithAnnotations GetFieldType(ConsList<FieldSymbol>
                             fieldsBeingBound = new ConsList<FieldSymbol>(this, fieldsBeingBound);
 
                             var initializerBinder = new ImplicitlyTypedFieldBinder(binder, fieldsBeingBound);
-                            var initializerOpt = initializerBinder.BindInferredVariableInitializer(diagnostics, RefKind.None, (EqualsValueClauseSyntax)declarator.Initializer, declarator);
+                            var executableBinder = new ExecutableCodeBinder(SyntaxNode, this, initializerBinder);
```

Comment:
I think `SyntaxNode` needs to be the initializer syntax, as passed to `BindInferredVariableInitializer` below.